### ISSUE TITLE
Warning: Cmodel got negative postshift!

### ISSUE THREAD

**Issue by whph1991** [rockchip-linux/rknn-toolkit#146]
I try to hrnetW18.onnx convert to rknn (onnx is normal when tested.)

There are problems with the conversion. What is the cause and how to fix it?

# Problems 

1. loading model step
  tensorflow errors
![image](https://user-images.githubusercontent.com/29689066/145663489-46fef8f8-10fa-455a-841d-224d603bc39e.png)

2. building step
  in verbose file, "D Infer onnx shape: Meet empty shape tensor, reshape () to (1,)!"
![image](https://user-images.githubusercontent.com/29689066/145663443-eb3bcda5-eb6f-4b7c-a2d6-31be6e4938a4.png)

3. running  model step
  no output and wanning
![image](https://user-images.githubusercontent.com/29689066/145663540-e1cb97bb-d191-4a16-aacd-085b70d0589c.png)

total terminal  print image
 
![image](https://user-images.githubusercontent.com/29689066/145663567-7272d083-d79a-41c4-ad21-b35de2a2328d.png)

# virtualenv

python: 3.6
opencv-python: 4.5.4.60
torch: 1.5.1
onnx: 1.6.0
rknn-toolkit: 1.7.1
tensorflow: 1.14.0
numpy: 1.16.3

# code

    import os
    import torch
    import numpy as np
    import cv2
    from rknn.api import RKNN
    
    ONNX_MODEL_NAME = 'hrnet_v1_6_0_ep_f.onnx'
    RKNN_MODEL = 'hrnet_v1_6_0_ep_f.rknn'
    IMG_PATH = os.path.join(os.getcwd(),'0001.jpg')
    
    def letterbox(img, height=608, width=1088,
                  color=(127.5, 127.5, 127.5)):  # resize a rectangular image to a padded rectangular
        shape = img.shape[:2]  # shape = [height, width]
        ratio = min(float(height) / shape[0], float(width) / shape[1])
        new_shape = (round(shape[1] * ratio), round(shape[0] * ratio))  # new_shape = [width, height]
        dw = (width - new_shape[0]) / 2  # width padding
        dh = (height - new_shape[1]) / 2  # height padding
        top, bottom = round(dh - 0.1), round(dh + 0.1)
        left, right = round(dw - 0.1), round(dw + 0.1)
        img = cv2.resize(img, new_shape, interpolation=cv2.INTER_AREA)  # resized, no border
        img = cv2.copyMakeBorder(img, top, bottom, left, right, cv2.BORDER_CONSTANT, value=color)  # padded rectangular
        return img, ratio, dw, dh
    
    if __name__ == '__main__':
    
        # Create RKNN object
        rknn = RKNN(verbose_file="rknn.txt")
        
        # Load ONNX model
        print('--> Loading model')
        ret = rknn.load_onnx(model=ONNX_MODEL_NAME)
    
        if ret != 0:
            print('Load hrnet failed!')
            exit(ret)
        print('done')
    
        # Build model
        print('--> Building model')
        ret = rknn.build(do_quantization=False)
        if ret != 0:
            print('Build hrnet failed!')
            exit(ret)
        print('done')
    
        # Export RKNN model
        print('--> Export RKNN model')
        ret = rknn.export_rknn(RKNN_MODEL)
        if ret != 0:
            print('Export hrnet.rknn failed!')
            exit(ret)
        print('done')
    
        print('--> init_runtime')
        ret = rknn.init_runtime()
        if ret != 0:
            print('Init runtime environment failed')
            exit(ret)
        print('done')
    
        img0 = cv2.imread(IMG_PATH)  # BGR
        assert img0 is not None, 'Failed to load ' + IMG_PATH
    
        # Padded resize
        img, _, _, _ = letterbox(img0)
    
        # Normalize RGB
        img = img[:, :, ::-1].transpose(2, 0, 1)
        img = np.ascontiguousarray(img, dtype=np.float32)
        img /= 255.0
    
        blob = torch.from_numpy(img).unsqueeze(0)
    
        input_img = blob.cpu().numpy()
    
        print('--> Running model')
        outputs = rknn.inference(inputs=[input_img])
        print('done')
    
        print(outputs)
    
        rknn.release()

**Comment by eRaul** [comment]
1. The warning of tensorflow can be ignored.
2. This model may have bugs in inferencing with the simulator, it is recommended to use the development board to inference.

**Comment by gioipv** [comment]
I met the same problem with my model. @whph1991 have u resolved this problem ?

**Comment by whph1991** [comment]
I solved problems!
Thanks!

Solution 1. Check Tensorflow version!
-. Bumps tensorflow from 1.14.0 to 2.5.2.
-. https://github.com/rockchip-linux/rknn-toolkit/pull/130

Solution 2. Change Model from hrnet to yolov5
-. if you want to use yolov5, see this example
-. https://github.com/rockchip-linux/rknn-toolkit/tree/master/examples/onnx/yolov5

Solution 3. Use the development board to inference!
-. no output warnning
